### PR TITLE
format dates as ISO8601

### PR DESCRIPTION
**Pull request by pyr**
first shot at implementing @jordansissel's request

### CODE REVIEW

File: src/main/java/net/logstash/log4j/JSONEventLayout.java [hunk 1]
```diff
@@ -88,4 +105,4 @@ private void addFieldData(String keyname, Object keyval){
             fieldData.put(keyname, keyval);
         }
     }
-}
\ No newline at end of file
+}
```
Comment (pyr) [inline src/main/java/net/logstash/log4j/JSONEventLayout.java:108]: sorry, emacs!

File: src/main/java/net/logstash/log4j/JSONEventLayout.java [hunk 2]
```diff
@@ -27,6 +29,21 @@
...
     private JSONObject logstashEvent;
 
+    public static String dateFormat(long timestamp) {
+	Date date = new Date(timestamp);
+	/*
+	 * SimpleDateFormat isn't thread safe so I need one 
+	 * instance per call, otherwise I'd have to pull in
+	 * joda time.
+	 */
+	SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```
Comment (jordansissel) [inline src/main/java/net/logstash/log4j/JSONEventLayout.java]: You should include high precision times, try making 'ss' into 'ss.SSS' to get at least milliseconds?

Comment (pyr) [inline src/main/java/net/logstash/log4j/JSONEventLayout.java]: agreed, see later commit

Comment (sjivan) [reply src/main/java/net/logstash/log4j/JSONEventLayout.java]: More efficient to use a ThreadLocal SimpleDateFormat

**Pull request merged**